lsp-types/src/Language/LSP/Protocol/Types/CustomInstances.hs

```diff
@@ -0,0 +1,27 @@
+{-# OPTIONS_GHC -Wno-orphans #-}
+module Language.LSP.Protocol.Types.CustomInstances where
+
+import           Language.LSP.Protocol.Internal.Types
+import           Data.Default
+import           Data.Hashable
+import           Data.Semigroup                ()
+
+instance Semigroup WorkspaceEdit where
+  (WorkspaceEdit a b c) <> (WorkspaceEdit a' b' c') = WorkspaceEdit (a <> a') (b <> b') (c <> c')
+instance Monoid WorkspaceEdit where
+  mempty = WorkspaceEdit Nothing Nothing Nothing
+
+instance Hashable Location
+instance Hashable Range
+instance Hashable Position
+
+instance Default ClientCapabilities where
```

Comment:
Suggest putting these in `Capabilities.hs`

Reply:
The reason why I didn't is because Capabilities doesn't export any of the Capability types, those are now all in Types . The advantage then of having them here is we can export instances with types.